homeassistant/components/mqtt/climate.py

```diff
@@ -447,19 +448,214 @@ async def _async_setup_entity(
+            )
+
+        self.add_subscription(
+            topics, CONF_CURRENT_TEMP_TOPIC, handle_current_temperature_received
+        )
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def handle_target_temperature_received(msg: ReceiveMessage) -> None:
+            """Handle target temperature coming via MQTT."""
+            self.handle_climate_attribute_received(
+                msg, CONF_TEMP_STATE_TEMPLATE, "_attr_target_temperature"
+            )
+
+        self.add_subscription(
+            topics, CONF_TEMP_STATE_TOPIC, handle_target_temperature_received
+        )
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def handle_temperature_low_received(msg: ReceiveMessage) -> None:
+            """Handle target temperature low coming via MQTT."""
+            self.handle_climate_attribute_received(
+                msg, CONF_TEMP_LOW_STATE_TEMPLATE, "_attr_target_temperature_low"
+            )
+
+        self.add_subscription(
+            topics, CONF_TEMP_LOW_STATE_TOPIC, handle_temperature_low_received
+        )
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def handle_temperature_high_received(msg: ReceiveMessage) -> None:
+            """Handle target temperature high coming via MQTT."""
+            self.handle_climate_attribute_received(
+                msg, CONF_TEMP_HIGH_STATE_TEMPLATE, "_attr_target_temperature_high"
+            )
+
+        self.add_subscription(
+            topics, CONF_TEMP_HIGH_STATE_TOPIC, handle_temperature_high_received
+        )
+
+        self._sub_state = subscription.async_prepare_subscribe_topics(
+            self.hass, self._sub_state, topics
+        )
+
+    async def _subscribe_topics(self) -> None:
+        """(Re)Subscribe to topics."""
+        await subscription.async_subscribe_topics(self.hass, self._sub_state)
+
+    async def _publish(self, topic: str, payload: PublishPayloadType) -> None:
+        if self._topic[topic] is not None:
+            await self.async_publish(
+                self._topic[topic],
+                payload,
+                self._config[CONF_QOS],
+                self._config[CONF_RETAIN],
+                self._config[CONF_ENCODING],
+            )
+
+    async def _set_climate_attribute(
+        self,
+        temp: float | None,
+        cmnd_topic: str,
+        cmnd_template: str,
+        state_topic: str,
+        attr: str,
+    ) -> bool:
+        if temp is None:
+            return False
+        changed = False
+        if self._optimistic or self._topic[state_topic] is None:
+            # optimistic mode
+            changed = True
+            setattr(self, attr, temp)
+
+        payload = self._command_templates[cmnd_template](temp)
+        await self._publish(cmnd_topic, payload)
+        return changed
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set hvac mode."""
```

Comment:
Why does this method not have an implementation?
Should we make this class an ABC and mark it with `@abstractmethod`?

Reply:
It should be removed. 👍